superset-frontend/src/dashboard/util/getPermissions.tsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+export default function getPermissions(
+  perm: string,
+  view: string,
+  roles: object,
+) {
+  const roleList = Object.entries(roles);
+  if (roleList.length === 0) return false;
+  let bool;
+
+  roleList.forEach(([role, permissions]) => {
+    bool = Boolean(
+      permissions.find(
+        (permission: Array<string>) =>
+          permission[0] === perm && permission[1] === view,
+      ),
+    );
+  });
+  return bool;
```

Comment:
```suggestion
  return roleList.filter(([role,perms]) => {
      return perms.filter(item => {
      	return item[0] === perm && item[1] === view
      }).length > 0;
  }).length > 0; 
```
I'm not sure if that's any easier to read, but I had to try...

Reply:
@pkdotson wanna check this out?

Reply:
Maybe it should be memoized too or stored in a map for faster lookup

Reply:
Yep will check out!